crate/operator/main.py

```diff
@@ -592,3 +593,45 @@ async def update_cratedb_resource(
                     await ensure_user_password_label(
                         core, namespace, user_spec["password"]["secretKeyRef"]["name"]
                     )
+
+
+@kopf.on.update("", "v1", "secrets", labels={LABEL_USER_PASSWORD: "true"})
+async def secret_update(
+    namespace: str,
+    name: str,
+    diff: kopf.Diff,
+    **kwargs,
+):
+    async with ApiClient() as api_client:
+        coapi = CustomObjectsApi(api_client)
+        core = CoreV1Api(api_client)
+
+        for operation, field_path, old_value, new_value in diff:
+            custom_objects = await coapi.list_namespaced_custom_object(
+                namespace=namespace,
+                group=API_GROUP,
+                version="v1",
+                plural=RESOURCE_CRATEDB,
+            )
+
+            for crate_custom_object in custom_objects["items"]:
+                host = await get_host(
+                    core, namespace, crate_custom_object["metadata"]["name"]
+                )
+
+                for user_spec in crate_custom_object["spec"]["users"]:
+                    if (
+                        user_spec["password"]["secretKeyRef"]["name"] == name
+                        and user_spec["password"]["secretKeyRef"]["key"] in field_path
+                    ):
```

Comment:
Regular K8s secrets' data is always in the `data` part of the resource. Which means, the password for a user is in `user_spec["password"]["secretKeyRef"]["key"]` underneath `data`. Hence, you can check for equality instead of using `in`

```suggestion
                    expected_field_path = ("data", user_spec["password"]["secretKeyRef"]["key"])
                    if (
                        user_spec["password"]["secretKeyRef"]["name"] == name
                        and field_path == expected_field_path
                    ):
```